app/models/notification.rb

```diff
@@ -1,13 +1,21 @@
+# frozen_string_literal: true
 
 class Notification < ApplicationRecord
-  def self.new_notification(sender_id, receiver_id)
-    notification = Notification.find_or_initialize_by(sender_id:, receiver_id:)
-    if notification.new_record?
-      notification.save
-    else
-      notification.increment!(:count)
-      notification.update(read: false)
-    end
-    notification.count
-  end
-end
\ No newline at end of file
+  def self.find_or_create_notification(sender_id, receiver_id)
+    Notification.find_or_create_by(sender_id:, receiver_id:)
+  end
+
+  def increment_total
+    update(total: total + 1)
+  end
+
+  def total_notifications
+    total
+  end
+
+  def self.clear_notifications(current_user_id, buyer_id, seller_id)
+    sender_id = current_user_id == seller_id ? buyer_id : seller_id
+    notification = find_by(receiver_id: current_user_id, sender_id:)
```

Comment:
1. Method name?
2. We should make an association in current_user for notifications using `reciever_id` as foreign_key
3. Scope